src/Pages/AuditLog/AuditLog.js

```diff
@@ -95,6 +116,49 @@ export default function AuditLogPage() {
       <h1 className='text-4xl font-extrabold leading-none tracking-tight text-gray-900 md:text-5xl lg:text-6xl dark:text-white mb-8'>
         Audit Logs
       </h1>
+      <div className='space-y-4'>
+        <div className='mt-6 p-4 bg-gray-800 rounded-lg border border-gray-700'>
+          <div className='grid grid-cols-1 md:grid-cols-2 gap-4 items-end'>
+            <div>
+              <label className='block text-sm font-medium text-gray-300 mb-2'>Search Users</label>
+              <input
+                className='w-full px-3 py-2 bg-gray-700 border border-gray-600 rounded-md text-white placeholder-gray-400 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-blue-500'
+                type='text'
+                placeholder='Search by first name, last name, or email'
+                value={searchQuery}
+                onChange={e => setSearchQuery(e.target.value)}
+                onKeyDown={event => {
+                  if (event.key === 'Enter') {
+                    if (currentPage !== 0) {
+                      setCurrentPage(0);
+                    } else {
+                      applyFilters();
+                    }
+                  }
+                }}
+              />
+            </div>
+            <FilterActivityTypes activityFilters={activityFilters} setActivityFilters={setActivityFilters} />
+            <div className='flex gap-2'>
+              <button
+                onClick={applyFilters}
+                className='px-4 py-2 bg-blue-600 text-white rounded-md hover:bg-blue-500 focus:outline-none focus:ring-2 focus:ring-blue-500'
+              >
+                Apply Filters
+              </button>
+              <button
+                onClick={clearFilters}
+                className='px-4 py-2 bg-gray-600 text-white rounded-md hover:bg-gray-500 focus:outline-none focus:ring-2 focus:ring-gray-500'
+              >
+                Clear Filters
+              </button>
+            </div>
+          </div>
+        </div>
+        {auditLogsData.items.map((log, index) => (
+          <AuditLogCard log={log} index={index} />
```

Comment:
do you need a key here?

idk if doing `key={JSON.stringify(log)}` is efficient